W3ChampionsStatisticService/Matches/CanceledMatchesHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using W3C.Domain.Repositories;
+using W3ChampionsStatisticService.Ports;
+using W3ChampionsStatisticService.ReadModelBase;
+using W3ChampionsStatisticService.Services;
+using W3C.Contracts.Matchmaking;
+using Serilog;
+using System.Transactions;
+using System;
+
+namespace W3ChampionsStatisticService.Matches;
+
+public class CanceledMatchesHandler(
+    IMatchEventRepository eventRepository,
+    IMatchRepository matchRepository,
+    ITransactionCoordinator transactionCoordinator,
+    ITrackingService trackingService) : IAsyncUpdatable
+{
+    private readonly IMatchEventRepository _eventRepository = eventRepository;
+    private readonly IMatchRepository _matchRepository = matchRepository;
+    private readonly ITransactionCoordinator _transactionCoordinator = transactionCoordinator;
+    private readonly ITrackingService _trackingService = trackingService;
+
+    public async Task Update()
+    {
+        var nextEvents = await _eventRepository.LoadCanceledMatches();
+
+        while (nextEvents.Count != 0)
+        {
+            foreach (var nextEvent in nextEvents)
+            {
+                try
+                {
+                    await using (var transaction = await AsyncTransactionScope.CreateAsync(_transactionCoordinator))
+                    {
+                        if (nextEvent.match.gameMode != GameMode.CUSTOM)
+                        {
+                            var ongoingMatch = await _matchRepository.LoadDetailsByOngoingMatchId(nextEvent.match.id);
+
+                            if (ongoingMatch == null)
+                            {
+                                Log.Warning($"Canceled match {nextEvent.match.id} not found");
+                            }
+                            else
+                            {
+                                if (ongoingMatch.Match != null)
```

Comment:
Let's not have 7 nested levels of logic

Reply:
I tried to keep the original logic of the other handlers to keep the diff "small".

Happy to rewrite all of them.

Reply:
I think if we're making some changes anyway, better to leave things better than we found them. I think it makes sense to improve things if we're already making changes to that code anyway. Can always split things in to separate commits to make PRs easier to review